core/tests/puppeteer/lighthouse_setup.js

```diff
@@ -268,6 +269,22 @@ const getSkillEditorUrl = async function(browser, page) {
   }
 };
 
+const generateDataForTopicAndStoryPlayer = async function(browser, page) {
+  try {
+    // eslint-disable-next-line dot-notation
+    await page.goto('http://127.0.0.1:8181/admin#/activities', { waitUntil: networkIdle });
+
+    await page.waitForSelector(generateTopicButton);
+    await page.click(generateTopicButton);
+
+    await page.waitForTimeout(20000);
```

Comment:
Is 20,000 not too much? I've not seen a timeout this big anywhere else in the repo

Reply:
Hi @kapilag96 thanks for pointing. Please take alook at the function `getStoryEditorUrl`, `getTopicEditorUrl`, `getExplorationEditorUrl`, there we have used `page.waitForTimeout` and also used 15000. For this use case the data takes a liitle while to get loaded, so I have used 15000.

Reply:
What are we waiting for with this timeout?

Reply:
We are waiting for the data of the topic and story to be generated in the admin page.

Reply:
And is there no way to know that the data was generated without waiting for a timeout? Where is the data stored?

Reply:
I think for now there is no way to know data was generated, if any error is there that will be caught. The data for the 2 topics are being generated from the admin page.

Reply:
We are setting a status message when this data is successsfully loaded? Can you check if we can use that instead of a timeout?

Reply:
Hi @kapilag96 I tried to find out a way to use that in this commit: b33242ca57e4d4328c9b2a3e48775b6ff4c181bb
But there might be sometiming issue still causing this test to fail. In my opinion 15,000 will work, as we have used that in the codebase for funtions in the same file (see getStoryEditorUrl, getTopicEditorUrl, getExplorationEditorUrl). 

![image](https://github.com/oppia/oppia/assets/96219910/73adbf1c-a6f9-44ce-a932-b3ec0f4927f4)